lib/linter.js

```diff
@@ -4,9 +4,21 @@ const stripBom = require('strip-bom');
 const EditorConfigResolver = require('./get-editor-config');
 let path = require('path');
 
+const MAX_AUTOFIX_PASSES = 10;
+
 const WARNING_SEVERITY = 1;
 const ERROR_SEVERITY = 2;
 
+/**
+ * @param {Object} options
+ * @param {string} options.source - The source code to fix.
+ * @param {Object[]} messages - The lint messages.
+ * @returns {string} output - The fixed source.
+ */
+function applyFixes(options /* messages */) {
```

Comment:
FWIW, I do think future iterations will want to move this back onto the `Linter.prototype` (because it will need things that already exist there, e.g. `.buildRule`), but that seems good to do when motivated by some failing tests 😸